lib/baldr/validator.rb

```diff
@@ -48,6 +48,10 @@ def validate_tree!(segment, grammar, structure)
   end
 
   def check_loop_count(loop, s)
+    if (s[:id] == 'NTE' && loop.count > s[:max])
```

Comment:
@npersad84 not exactly what I meant. 
1. pass an extra value here - https://github.com/cloudlogistics/cloudlogistics/blob/production/broker_app/app/services/envelope_preparation_service.rb#L8
2. which comes to this file, and then will do the same generically for any loop